### PR TITLE
Add `max_open_files` parameter for systemd systems

### PR DESCRIPTION
**Pull request by alexjfisher**
Alternative to https://github.com/voxpupuli/puppet-prometheus/pull/297

@sc0rp10 Does this work for you?

### CODE REVIEW

**Comment by bastelfreak** [comment]
@alexjfisher ping :)

File: manifests/init.pp [hunk 1]
```diff
@@ -180,6 +185,7 @@
   Hash $config_defaults = {},
   String $os            = downcase($facts['kernel']),
   Optional[Variant[Stdlib::HTTPUrl, Stdlib::Unixpath, String[1]]] $external_url = undef,
+  Optional[Integer] $max_open_files = undef,
```
Comment (dhoppe) [inline manifests/init.pp:202]: @alexjfisher Could you please fix the indentation?

**Comment by bastelfreak** [comment]
Ping @alexjfisher

File: manifests/init.pp (hunk 1 above)
Comment (alexjfisher) [reply manifests/init.pp:202]: I didn't think adding 45 extra whitespaces improved readability.

Comment (alexjfisher) [reply manifests/init.pp:202]: Rebased (which added the `collect_scrape_jobs` line above mine)

**Pull request merged**